### PR TITLE
Refactor tenant-volume -volume flashcopy mapping data source and documentation

### PR DESCRIPTION
**Pull request by Diptipowervs**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Relates OR Closes #0000

Output from acceptance testing:

```
$ make testacc TESTARGS='-run=TestAccXXX'

...
```

### CODE REVIEW

File: ibm/service/power/data_source_ibm_pi_volume.go [hunk 1]
```diff
@@ -125,32 +128,31 @@ func dataSourceIBMPIVolumeRead(ctx context.Context, d *schema.ResourceData, meta
...
-	d.Set("replication_type", volumedata.ReplicationType)
-	d.Set("replication_status", volumedata.ReplicationStatus)
-	d.Set("auxiliary", volumedata.Auxiliary)
-	d.Set("consistency_group_name", volumedata.ConsistencyGroupName)
-	d.Set("group_id", volumedata.GroupID)
-	d.Set("mirroring_state", volumedata.MirroringState)
-	d.Set("primary_role", volumedata.PrimaryRole)
-	d.Set("auxiliary_volume_name", volumedata.AuxVolumeName)
-	d.Set("master_volume_name", volumedata.MasterVolumeName)
-	d.Set("io_throttle_rate", volumedata.IoThrottleRate)
```
Comment (yussufsh) [inline ibm/service/power/data_source_ibm_pi_volume.go:153]: Why `io_throttle_rate` is removed?

**Review by yussufsh — CHANGES_REQUESTED** [review]

File: ibm/service/power/data_source_ibm_pi_volume.go [hunk 2]
```diff
@@ -6,114 +6,117 @@ package power
...
+			// Arguments
+			Arg_CloudInstanceID: {
+				Description:  "The GUID of the service instance associated with an account.",
 				Required:     true,
-				Description:  "Volume Name to be used for pvminstances",
+				Type:         schema.TypeString,
 				ValidateFunc: validation.NoZeroValues,
 			},
-			helpers.PICloudInstanceId: {
+			helpers.PIVolumeName: {
```
[the remark this answers is not in the document]
Comment (ismirlia) [reply ibm/service/power/data_source_ibm_pi_volume.go:28]: yes, i'll make the change. This one's an oversight.

File: ibm/service/power/data_source_ibm_pi_volume.go (hunk 1 above)
Comment (ismirlia) [reply ibm/service/power/data_source_ibm_pi_volume.go:153]: This was a merge error. I'll fix it also.

**Review by yussufsh — APPROVED** [review]
lgtm

**Review by michaelkad — APPROVED** [review]
Lgtm

**Review by ismirlia — APPROVED** [review]

**Review by hkantare — APPROVED** [review]

**Pull request merged**